zarf.yaml

```diff
@@ -50,6 +50,15 @@ components:
     actions:
       onDeploy:
         before:
+          - cmd: |
+              k3d_version=$(k3d version | awk '/k3d version/ {print $3}' | sed 's/^v//')
+              required_version="5.7.1"
+              if [ "$(printf "%s\n%s" "$required_version" "$k3d_version" | sort -V | head -n 1)" != "$required_version" ]; then
+                echo "This package requires a minimum k3d version of $required_version"
+                echo "Please upgrade k3d (https://k3d.io/stable/#install-current-latest-release) and try again"
+                exit 1
+              fi
+            description: "Check k3d version compatibility"
```

Comment:
Open for discussion on whether we want to do this. I think technically an older k3d version would work, if you also override to use an older k3s image. By default my experience is that it would fail to deploy (since we use 1.30/1.31 by default) with anything pre-5.7.1.

This is arbitrarily adding a new constraint that mostly makes sense but the check is a bit janky becuase I tried to keep it to standard utilities available on both linux and mac machines.

Reply:
After discussion it seems like this is preferable to just docs, and I made some changes to hopefully simplify the checks here.